apps/nextra/pages/en/build/guides/aptos-keyless/how-keyless-works.mdx

```diff
@@ -0,0 +1,87 @@
+
+**Step 4**: Once the user has signed in, Google sends the dApp a signed JWT, which includes the user's `sub` identifier (e.g., `uid-123`), the application’s `aud` identifier (e.g., `"dapp-xyz"`) and the `nonce` with the EPK commitment. (This assumes that the dApp has previously registered with Google and received this `"dapp-xyz"` identifier.)
+
+**Step 5**: The dApp now has almost everything it needs to derive a keyless account for the user: the user’s identifier (`sub`) and the dApp’s identifier (`aud`). But, to preserve the privacy of the user, the dApp will use a third piece of information: a blinding factor $r$ called a **pepper**. The dApp will contact a so-called **guardian** who will deterministically derive a random $r$ from the given (`sub`, `aud`). Importantly, the guardian will only reveal $r$ to the dApp upon seeing a validly-signed JWT for the queried (`sub`, `aud`).
+
+**Step 6**: The dApp derives the address of the account as $\mathsf{addr} = H(\texttt{"uid-123"}, \texttt{"dapp-xyz"}, r)$, where $H$ is a cryptographic hash function.
+
+Note that the pepper $r$ is used to hide the user and app identity inside the address since, as we described above, only an authorized user with a valid JWT will be able to obtain this pepper.
+
+Also, note that the address is independent of the EPK. This is why the ESK need not be long-lived and can be lost.
+
+Finally, the dApp can, for example, send an NFT to the user at their address $\mathsf{addr}$.
+
+But how can the dApp authorize TXN from this account at $\mathsf{addr}$? We discuss that next.
+
+## Flow: Obtaining a zero-knowledge proof before transacting
+
+In the previous flow, we showed how a dApp can sign in a Google user and derive their privacy-preserving keyless address, with the help of a guardian.
+
+Next, we show how this dApp can obtain a zero-knowledge proof (ZKP), which will allow it to authorize transactions from this address for the user. Importantly, the transaction will hide the user’s identifying information (e.g., the `sub` field).
+
+![Keyless proof diagram](/docs/aptos-keyless/keyless-proof.png "Keyless proof diagram")
+
+**Step 1**: The dApp sends all the necessary public information (i.e., $\mathsf{epk}$, $\mathsf{GPK}$) and private information (i.e., JWT, signature $\sigma_G$ from Google, EPK blinding factor $\rho$, and pepper $r$) to the **prover service**.
+
+**Step 2**: The prover derives the user’s address $\mathsf{addr}$ and computes a zero-knowledge proof (ZKP) $\pi$ for the keyless relation $\mathcal{R}_\mathsf{keyless}$ (described below). This proof acts as a **privacy-preserving** digital certificate, and binds the user's address $\mathsf{addr}$ to the ephemeral public key $\mathsf{epk}$. The prover then sends $\pi$ to the dApp.
```

Comment:
Steps component

Reply:
I think for this i prefer no step component, as these are not actual steps for the user to take.  don't want to add to the right hand nav bar